grid/social-media/publish-post/maps/pinterest.suma

```diff
@@ -0,0 +1,118 @@
+profile = "social-media/publish-post@1.1"
+provider = "pinterest"
+
+// https://developers.pinterest.com/docs/api/v5/#operation/pins/create
+map PublishPost {
+  return map error if (!Array.isArray(input.media) || input.media.length !== 1) {
+    title = "Invalid media"
+    detail = "You must provide exactly one media item for Pinterest."
+  }
+
+  // FIXME[linter]: Workaround for `ElementAccessExpression - Wrong Input Structure: expected {profileId: string!, text: string!, link: string!, media: [{url: string!, contents: any, altText: string}]}, but got input.media.0`
+  set {
+    origMedia = input.media
+  }
+  
+  set {
+    media = origMedia[0]
+    mediaSource = {}
+  }
+
+  set if (media.contents) {
+    mediaSource = {
+      source_type: 'image_base64',
+      data: Buffer.isBuffer(media.contents) ? media.contents.toString('base64') : media.contents,
+      content_type: undefined,
+    }
+  }
+  // https://stackoverflow.com/a/69570564 :wow:
+  set if (mediaSource.data && mediaSource.data.startsWith('/9j')) {
+    mediaSource.content_type = 'image/jpeg'
+  }
+  set if (mediaSource.data && mediaSource.data.startsWith('iVB')) {
+    mediaSource.content_type = 'image/png'
+  }
+
+  set if (media.url) {
+    mediaSource = {
+      source_type: 'image_url',
+      url: media.url,
+    }
+  }
+
+  // return map error if (!mediaSource.url && !mediaSource.content_type) {
```

Comment:
Why is this commented out? Does Pinterest accept media without defined `content_type`?

Reply:
Good point, I have omitted this part with intention to test it first, but forgot about it.